src/ApiService/ApiService/onefuzzlib/ScalesetOperations.cs

```diff
@@ -122,6 +122,8 @@ public ScalesetOperations(ILogTracer log, IOnefuzzContext context)
                 _log.WithHttpStatus(r.ErrorV).Error($"Failed to save scaleset record {scaleset.ScalesetId}");
             }
         }
+
+        return scaleset with  { State = ScalesetState.Halt }; 
```

Comment:
the state should already be halt at this point  so you should be returning the object as is